uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/controllers/request.controller.js

```diff
@@ -37,198 +39,127 @@ define([
     var initialLeaveRequestAttributes = {}; // used to compare the change in leaverequest in edit mode
     var listeners = [];
     var loggedInContactId = '';
-    var role = '';
-    var vm = this;
     var NO_ENTITLEMENT_ERROR = 'No entitlement';
+    var role = '';
+    var vm = _.assign(this, directiveOptions); // put all directive options directly in the vm
```

Comment:
Any chance that directive options intersect with vm? For example, what if both have props like `id` or `mode`?